wgpu-hal/src/dx12/command.rs

```diff
@@ -228,8 +228,23 @@ impl crate::CommandEncoder<super::Api> for super::CommandEncoder {
     unsafe fn begin_encoding(&mut self, label: crate::Label) -> Result<(), crate::DeviceError> {
         let list = match self.free_lists.pop() {
             Some(list) => {
-                list.reset(self.allocator, native::PipelineState::null());
-                list
+                if list
+                    .reset(self.allocator, native::PipelineState::null())
+                    .into_result()
+                    .is_ok()
```

Comment:
Could we move this into it's own variable

Reply:
I'm so sorry I can't understand exactly what you mean.

Reply:
Basically:

```rust
let reset_result = list
                    .reset(self.allocator, native::PipelineState::null())
                    .into_result();
if reset_result.is_ok() {
...
}
```

To get the bulk of the code out of the if statement condition.

Reply:
Got it!

Reply:
I refactored the logic of reusing the list, we can always try to get the next reusable list instead of once.